src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3025,6 +3027,20 @@ namespace winrt::TerminalApp::implementation
         term.RefreshQuickFixMenu();
     }
 
+    winrt::fire_and_forget TerminalPage::_WindowSizeChanged(const IInspectable /*sender*/, const Microsoft::Terminal::Control::WindowSizeChangedEventArgs args)
+    {
+        co_await wil::resume_foreground(Dispatcher());
```

Comment:
Since you already switched to the UI thread in `TermControl` you won't need to do it again here.

Oher functions in this file do it, because our event handling code hasn't been cleaned up in a while and so some functions switch to the UI thread here and some in `TermControl`. I think it's better to do it in `TermControl`, because that makes it safer and more consistent.